general:
  world_frame: "odom"
  publish_performance_stats: false

map:
  pruning_period: { seconds: 10.0 }
  visualization_period: { seconds: 10.0 }
  autosave_period: { seconds: -1.0 }
  autosave_path: ""
  data_structure:
    type: "octree"
    min_cell_width: &min_cell_width { meters: 0.02 }
    max_height: 14

integrator:
  - input:
      type: "depth_camera"
      topic_name: "/data/depth_image"
      topic_queue_length: 10
      max_wait_for_pose: { seconds: 1.0 }
    integration_method:
      type: "coarse_to_fine_integrator"
      min_range: { meters: 0.1 }
      max_range: { meters: 5.0 }
      max_acceptable_update_error: 0.1
      max_update_resolution: *min_cell_width
    projection_model:
      type: "pinhole_camera_projector"
      width: 640
      height: 480
      fx: 320.0
      fy: 320.0
      cx: 320.0
      cy: 240.0
    measurement_model:
      type: "continuous_volumetric_log_odds"
      angle_sigma: { radians: 0.3 }
      range_sigma: { meters: 0.03 }
      scaling_free: 0.2
      scaling_occupied: 0.4
